src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/ConnectionButton.java

```diff
@@ -0,0 +1,135 @@
+package com.marginallyclever.makelangelo.plotter.plotterControls;
+
+import com.marginallyclever.communications.*;
+import com.marginallyclever.convenience.ButtonIcon;
+import com.marginallyclever.makelangelo.Translator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * ConnectionButton provides a human interface to open or close a
+ * connection to a remote device available through a {@link NetworkSession}.
+ *
+ * @author Dan Royer
+ * @since 7.28.0
+ */
+public class ConnectionButton extends JPanel {
+	private static final Logger logger = LoggerFactory.getLogger(ConnectionButton.class);
+	
+	private static final long serialVersionUID = 4773092967249064165L;
+	@Deprecated
+	public static final int CONNECTION_OPENED = 1;
+	@Deprecated
+	public static final int CONNECTION_CLOSED = 2;
+	
+	private ButtonIcon bConnect = new ButtonIcon("ButtonConnect", "/images/connect.png");
```

Comment:
This is the connect button.  The rest is part of the job of choosing a connection.  That's why I preferred "ChooseConnection".

Reply:
ok, it is now done